### PR TITLE
Add swagger documentation

### PR DESCRIPTION
**Pull request by Purhan**
# Description
Added swagger API docs using drf-yasg. Preview below:
image

## Related issues
Please add related issues.
#323

## Type of change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue).
- [x] New feature (non-breaking change which adds functionality).
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected).

# Checklist

- [x] The pull request is for the branch develop
- [ ] If I added a new analyzer, I updated the file [Usage](https://github.com/intelowlproject/IntelOwl/blob/master/docs/source/Usage.md). If the analyzer provides additional optional configuration, I added the available options here: [Advanced-Usage](./Advanced-Usage.md)
- [ ] If I added external libraries/packages that use restrictive licenses, please add them in the [ReadMe - Legal Notice](https://github.com/certego/IntelOwl/blob/master/README.md) section
- [ ] I added new secrets in the files [env_file_app_template](https://github.com/intelowlproject/IntelOwl/blob/master/env_file_app_template), [env_file_app_ci](https://github.com/certego/IntelOwl/blob/master/env_file_app_travis) and in the docs: [Installation](./Installation.md)
- [x] I have added tests in the [Tests](https://github.com/intelowlproject/IntelOwl/blob/master/tests) folder. 
- [x] The tests gave 0 errors.
- [x] `Black` gave 0 errors.
- [x] `Flake` gave 0 errors.
- [x] I squashed the commits into a single one.
  
### please follow these rules
- If your changes decrease the overall tests coverage (you will know after the Codecov CI job is done), you should add the required tests to fix the problem
- Everytime you make changes to the PR and you think the work is done, you should explicitly ask for a review

### CODE REVIEW

**Review by mlodic — COMMENTED** [review]
thank you for the PR!

It is possible to set up the authorization via API token? I ask this because ATM it is only possible to test the APIs from the swagger docs by only using the Basic Auth that is not supported by IntelOwl

image

Also I think we should rewrite the docstrings of the APIs because swagger cannot parse that data correctly. Ex:
image

File: intel_owl/settings.py [hunk 1]
```diff
@@ -12,7 +12,7 @@
 SECRET_KEY = os.environ.get("DJANGO_SECRET", None) or get_random_secret_key()
 
 # SECURITY WARNING: don't run with debug turned on in production!
-DEBUG = os.environ.get("DEBUG", False) == "True"
+DEBUG = True
```
[the remark this answers is not in the document]
Comment (Purhan) [reply intel_owl/settings.py:15]: omg how did I miss this :laughing:, will revert it soon.

**Comment by Purhan** [comment]
> Is it possible to set up the authorization via API token? I ask this because ATM it is possible to test the APIs from the swagger docs only by the Basic Auth that is not supported by IntelOwl

Yes, we can use any permission class from the Rest-API framework.

**Comment by Purhan** [comment]
I have reverted the unrelated changes, added an authentication class so that it can't be accessed without the authentication token. Now what's left is changing the docstrings. Will update soon.

**Comment by mlodic** [comment]
Am I me only or you get "401 Unauthorized" too when going to `/api/docs`?

**Comment by Eshaan7** [comment]
Hi. Thankyou for your contribution. However, there are a few caveats in the way you implemented this is very different from our use case.

If you see #332 and  [this comment](https://github.com/intelowlproject/IntelOwl/pull/332#issuecomment-770382482)., someone already did this and we had to close that PR too for same reason.

> Use [this](https://drf-yasg.readthedocs.io/en/stable/rendering.html#management-command) to create the spec file, and then use the `sphinx-redoc` extension to generate redoc along with rest of the docs. Then link both docs together with [something like this](https://github.com/sphinx-contrib/redoc/issues/15). There should be a GH workflow setup for this.

**Comment by Purhan** [comment]
I see, thank you for reviewing. I'm sorry but I can't work on this anymore due to personal engagements. Closing this for now. Thank you for your time @Eshaan7 @mlodic :+1:

**Comment by Eshaan7** [comment]
> Also I think we should rewrite the docstrings of the APIs because swagger cannot parse that data correctly.

Yes, it's our mistake for not following a proper syntax while writing doc strings. I like the [napoleon syntax](https://sphinxcontrib-napoleon.readthedocs.io/en/latest/).

Need to use proper syntax as given [here](https://drf-yasg.readthedocs.io/en/1.0.4/custom_spec.html).
> descriptions for Operations, Parameters and Schemas are picked up from docstrings and help_text attributes in the same manner as the default DRF SchemaGenerator

**Pull request closed without merging**